src/components/__tests__/use-auth.test.ts

```
@@ -0,0 +1,234 @@
+})
+
+// Mock global fetch
+global.fetch = vi.fn()
+
+describe('useAuth', () => {
+  const mockApiCredentials = { username: 'test', password: 'pass' }
+
+  beforeEach(() => {
+    vi.clearAllMocks()
+    mockEnv.VITE_STORAGE_MODE = 'local'
+    ;(useLocalStorage as any).mockReturnValue([mockApiCredentials])
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)